public/components/reindexController.react.js

```diff
@@ -37,10 +40,19 @@ export default class ReindexControllerComponent extends React.Component {
             this.loadContentSource(nextProps.routeParams.id, nextProps.routeParams.environment);
             this.loadReindexHistory(nextProps.routeParams.id, nextProps.routeParams.environment);
             this.loadRunningReindex(nextProps.routeParams.id, nextProps.routeParams.environment);
+            this.loadContentSourceWithId(nextProps.routeParams.id);
             this.setState({editModeOn: false});
         }
     }
 
+    loadContentSourceWithId(id) {
+        ContentSourceService.getContentSourcesWithId(id).then(response => {
+            this.setState({
+                contentSourcesForEnvironments: response.contentSources
+            });
+        });
+    }
+
     loadContentSource(id, environment) {
```

Comment:
But if you are loading both a single content source on mount and ALL content sources on mount then can't you do something clever on mount to select the one you want (based on the props) from the ones you have loaded with the environments?